### PR TITLE
Add Jetty Threadpool Readiness Health Checks

### PR DESCRIPTION
**Pull request by kevink-sq**
This change adds readiness health checks to the Jetty threadpool to signal the degraded state.

edit: had to run `gradle apiDump` to generate config artifacts similar to https://github.com/cashapp/misk/pull/2921/files and https://github.com/cashapp/misk/pull/3134/files

### CODE REVIEW

File: misk/src/main/kotlin/misk/web/MiskWebModule.kt [hunk 1]
```diff
@@ -298,6 +300,7 @@ class MiskWebModule @JvmOverloads constructor(
       }
       bind<ThreadPool>().toInstance(threadPool)
       bind<MeasuredThreadPool>().toInstance(MeasuredThreadPoolExecutor(executor))
+      multibind<HealthCheck>().to<JettyThreadPoolHealthCheck>()
```
Comment (ThePumpingLemma) [inline misk/src/main/kotlin/misk/web/MiskWebModule.kt:303]: Don't we want this for `QueuedThreadPool` as well?

Comment (ThePumpingLemma) [reply misk/src/main/kotlin/misk/web/MiskWebModule.kt:303]: Also, maybe we should make this configurable so we can explicitly opt into it and test it out?

Comment (tirsen) [reply misk/src/main/kotlin/misk/web/MiskWebModule.kt:303]: yeah adding health checks can be a bit risky as they can cause flapping and sitting ducks, roll it out very carefully

Comment (kevink-sq) [reply misk/src/main/kotlin/misk/web/MiskWebModule.kt:305]: Made the health checks configurable and fixed to include `QueuedThreadPool`

**Review by ThePumpingLemma — APPROVED** [review]

**Pull request merged**